### PR TITLE
Added validation for manifest types (#839)

### PR DESCRIPTION
**Pull request by GPelayo**

### CODE REVIEW

File: lambdas/service/app.py [hunk 1]
```diff
@@ -421,12 +421,18 @@ def get_manifest():
...
-    format = params.get('format', 'tsv')
     es_td = EsTd()
-    response = es_td.transform_manifest(format, filters)
+    response = es_td.transform_manifest(get_format(params), filters)
     return response
 
 
+def get_format(params):
+    format_ = params.get('format', 'tsv')
+    if format_ not in {'tsv', 'bdbag'}:
```
Comment (hannes-ucsc) [inline lambdas/service/app.py]: Eliminate `not`, use `return` in `if` branch and `raise` in `else` branch.

Comment (hannes-ucsc) [inline lambdas/service/app.py]: Nit: 

```suggestion
    if format_ not in ('tsv', 'bdbag'):
```

File: lambdas/service/app.py [hunk 2]
```diff
@@ -457,6 +463,9 @@ def start_manifest_generation():
     If the manifest generation is done and the manifest is ready to be downloaded, the response will
     have a 302 status and will redirect to the URL of the manifest.
     """
+    query_params = app.current_request.query_params
```
Comment (hannes-ucsc) [inline lambdas/service/app.py]: Inline.

File: test/service/test_request_validation.py [hunk 3]
```diff
@@ -487,3 +487,10 @@ def test_bdbag_manifest(self):
                     'fastq[read2].dos_url',
                     'fastq[read2].file_url',
                 ], reader.fieldnames)
+
+    def test_manifest_format_validation(self):
+        for manifest_endpoint in '/manifest/files', '/repository/files/export':
+            with self.subTest(manifest_endpoint=manifest_endpoint):
+                url = self.base_url + f'{manifest_endpoint}?format=invalid-type'
+                response = requests.get(url)
+                self.assertEqual(400, response.status_code, response.content)
```
Comment (hannes-ucsc) [inline test/service/test_request_validation.py]: Why do you use the `response.content` as the assertion error message? Please explain.

Comment (GPelayo) [reply test/service/test_request_validation.py]: While coding this test, this assert kept getting 500 response errors. Adding `response.content` as the assertion error message helped me diagnose the source of those errors.

**Comment by hannes-ucsc** [comment]
Sandbox was poisoned by a previous PR.

**Comment by hannes-ucsc** [comment]
Fixed sandbox, now retrying build.

**Pull request merged**